### PR TITLE
add multiplayer() and update some comments in terms of Doxygen.

### PR DESCRIPTION
**Pull request by JeanneXu1122**

### CODE REVIEW

**Review by Pari-Firoozi — CHANGES_REQUESTED** [review]
Thanks Jin. 
I took a quick look and realized that you are not working on the latest version. I saw you include some file like player.h which i rename it in the previous commits. Can you please rebase your branch. Then I will review.
Thanks.

**Comment by JeanneXu1122** [comment]
I did create this branch based on the latest version of dev. And I just changed a little bit for the include part of take _heart.c which as you mentioned was adjusted before. So I only need to revert the change I made, and then you can review the code mainly related to the multiplayer function.

File: include/map.h [hunk 1]
```diff
@@ -109,6 +109,18 @@ typedef struct map_textures {
     SDL_Texture* p_texture_barrier;
 } map_textures_t;
 
+typedef struct map_textures_multi {
```
Comment (Pari-Firoozi) [inline include/map.h:112]: Can't you add the new textures to the previous one and use the same texture?

**Review by Pari-Firoozi — CHANGES_REQUESTED** [review]
I couldn't able to compile the codes.
Please update makefile as you add new files so we can compile and run that.

File: include/map.h (hunk 1 above)
Comment (layeghimahsa1996) [reply include/map.h:112]: Yes I agree

**Review by layeghimahsa1996 — COMMENTED** [review]
Thanks Jin.
It looks good. Just follow the latest version so that you don't miss anything.

**Review by Pari-Firoozi — CHANGES_REQUESTED** [review]
Thanks Jin for the update. Seems ok. 
Just because in programing we should avoid repetitions, we can have just one generate view  function and add a for loop for player and its dependencies as well, like what you did for barrier and arrows( we know the number of players: p_map->number_of_players ).

Thanks.

**Comment by JeanneXu1122** [comment]
I tried to update code as you suggested, but it turned out not as expected. And as for players, they have different attributes for themselves, so I think it's acceptable to implement the two functions separately.

**Comment by Pari-Firoozi** [comment]
Ok, lets merge it to see how it look likes and we can improve that later.

**Review by Pari-Firoozi — APPROVED** [review]

**Review by layeghimahsa1996 — APPROVED** [review]

**Review by Jaserelhabrouk — APPROVED** [review]
Looks good. Thank you

**Pull request merged**